presto-orc/src/main/java/com/facebook/presto/orc/OrcWriterOptions.java

```diff
@@ -58,7 +65,10 @@ private OrcWriterOptions(
             OptionalInt compressionLevel,
             StreamLayout streamLayout,
             boolean integerDictionaryEncodingEnabled,
-            boolean stringDictionarySortingEnabled)
+            boolean stringDictionarySortingEnabled,
+            boolean dwrfStripeCacheEnabled,
```

Comment:
Yes. But, if I understand it correctly, if we extract these fields into a top level object then it would not follow the way these Reader/WriterOptions are built by OrcFileWriterConfig, which applies one config value a time. So, that's the reason I kept it flat.

```java
    public int getStripeMaxRowCount()
    {
        return stripeMaxRowCount;
    }

    @Config("hive.orc.writer.stripe-max-rows")
    public OrcFileWriterConfig setStripeMaxRowCount(int stripeMaxRowCount)
    {
        this.stripeMaxRowCount = stripeMaxRowCount;
        return this;
    }
```

That brings me to the point, that dwrf and orc have diverged significantly, and it would be nice to completely split them into different modules at some point.

Reply:
It seems we would need a `toDwrfWriterOptionsBuilder` 🤔 I think it would be useful to separate shared config and dwrf specific config so it is easier when we decide to split them into different modules.

Reply:
I need a little bit more input on how you see it. My current understanding is that I should create extract dwrf* fields into a new classDwrfWriterOptions. Do you see this class to be a field of OrcWriterOptions, or do you prefer it to be  passed in into the OrcWriter as a separate constructor argument?

Reply:
* Extracted DWRF specific options for this PR into a separate class
* added a test for the class
* reverted changes to the OrcWriterOptions
* updated PR description